packages/api/src/apps/datastore_api/app.ts

```diff
@@ -154,57 +157,21 @@ export default async (env: ProcessEnv) => {
     globalStats: Services.stats.Global({ debug }, appState),
   };
 
-  const initBlock = await provider.getBlock("latest");
-
-  // warm caches
-  await services.registry(appState.lastBlockUpdate, initBlock.number);
-  console.log("Got all EMP addresses");
-
-  await services.lspCreator.update(appState.lastBlockUpdate, initBlock.number);
-  console.log("Got all LSP addresses");
-
-  await services.emps.update(appState.lastBlockUpdate, initBlock.number);
-  console.log("Updated EMP state");
-
-  await services.lsps.update(appState.lastBlockUpdate, initBlock.number);
-  console.log("Updated LSP state");
-
-  // we've update our state based on latest block we queried
-  appState.lastBlockUpdate = initBlock.number;
-
-  await services.erc20s.update();
-  console.log("Updated tokens");
-
   // backfill price histories, disable if not specified in env
   if (env.backfillDays) {
     console.log(`Backfilling price history from ${env.backfillDays} days ago`);
     await services.collateralPrices.backfill(moment().subtract(env.backfillDays, "days").valueOf());
     console.log("Updated Collateral Prices Backfill");
+
+    // backfill price history only if runs for the first time
+    if (await appState.appStats.getLastBlockUpdate()) return;
```

Comment:
i dont think we can just return early here because the rest of the app needs to init. maybe we just add this check to line 161 to not enter this block if it exists

Reply:
I replaced the return with a conditional check. Wasn't my intention to stop the execution of the program 😄 One mention tho, collateralPrices are still kept in memory so those ones need to be backfilled